pages/transactions/[id].js

```diff
@@ -210,11 +222,116 @@ export default function TransactionDetailsPage({
             </StyledOptionsContainer>
           </StyledDetailsContainer>
         )}
+
+        {isDeletingNote ? (
+          <StyledConfirmActionContainer>
+            <StyledCancelButton
+              type="button"
+              onClick={() => setIsDeletingNote(false)}
+            >
+              Cancel
+            </StyledCancelButton>
+            <StyledConfirmButton
+              type="button"
+              onClick={() => {
+                handleConfirmDeleteNote();
+              }}
+            >
+              Really Delete
+            </StyledConfirmButton>
+          </StyledConfirmActionContainer>
+        ) : (
+          <StyledNoteArea>
+            Notes
+            {isNoteEdit ? (
+              <StyledTextArea
+                id="note"
+                name="note"
+                defaultValue={transactionDetails.note}
+                autoFocus
+                maxLength={140}
+                onKeyDown={handleNoteUpdate}
+                onBlur={() => setIsNoteEdit(false)}
+              />
+            ) : (
+              <StyledNoteContentButton
+                onClick={() => setIsNoteEdit(true)}
+                empty={!transactionDetails.note}
+              >
+                {transactionDetails.note}
+              </StyledNoteContentButton>
+            )}
+            {isNoteError && (
+              <ErrorMessageNote>
+                Please enter at least 1 valid character (letter, number or
+                special character) to submit.
+              </ErrorMessageNote>
+            )}
+            <StyledOptionsContainer>
+              <StyledDeleteButton
+                type="button"
+                onClick={() => {
+                  setIsDeletingNote(true);
+                }}
+              >
+                <StyledImage
+                  aria-hidden="true"
+                  src="/images/trash.svg"
+                  alt="delete button"
+                  width={15}
+                  height={15}
+                />
+              </StyledDeleteButton>
+            </StyledOptionsContainer>
+          </StyledNoteArea>
+        )}
       </StyledTransactionDetails>
+      {successMessage && (
+        <StyledSuccessMessage>{successMessage}</StyledSuccessMessage>
+      )}
     </>
   );
 }
 
+const StyledNoteContentButton = styled.button`
+  border: none;
+  background-color: inherit;
+  min-height: 158px;
+  display: flex;
+  padding: 0;
+  text-align: left;
+  font-size: 1rem;
+  line-height: 1.5;
+  padding-top: 10px;
+  padding-left: 2px;
+  font-family: inherit;
+
+  &::after {
+    content: ${(props) => (props.empty ? '"Click to add a note"' : '""')};
+    color: grey;
+  }
```

Comment:
using css to set the content of an element is a bit of an anti pattern. Please define the default message in the jsx above like `<button>{note || "Click to add note."}</button>`